src/ImageSharp/Formats/Webp/WebpChunkParsingUtils.cs

```diff
@@ -0,0 +1,351 @@
+        /// - An optional 'ALPH' chunk with alpha channel data.
+        /// After the image header, image data will follow. After that optional image metadata chunks (EXIF and XMP) can follow.
+        /// </summary>
+        /// <returns>Information about this webp image.</returns>
+        public static WebpImageInfo ReadVp8XHeader(Stream stream, byte[] buffer, WebpFeatures features)
+        {
+            uint fileSize = ReadChunkSize(stream, buffer);
+
+            // The first byte contains information about the image features used.
+            byte imageFeatures = (byte)stream.ReadByte();
+
+            // The first two bit of it are reserved and should be 0.
+            if (imageFeatures >> 6 != 0)
+            {
+                WebpThrowHelper.ThrowImageFormatException("first two bits of the VP8X header are expected to be zero");
+            }
+
+            // If bit 3 is set, a ICC Profile Chunk should be present.
+            features.IccProfile = (imageFeatures & (1 << 5)) != 0;
+
+            // If bit 4 is set, any of the frames of the image contain transparency information ("alpha" chunk).
+            features.Alpha = (imageFeatures & (1 << 4)) != 0;
+
+            // If bit 5 is set, a EXIF metadata should be present.
+            features.ExifProfile = (imageFeatures & (1 << 3)) != 0;
+
+            // If bit 6 is set, XMP metadata should be present.
+            features.XmpMetaData = (imageFeatures & (1 << 2)) != 0;
+
+            // If bit 7 is set, animation should be present.
+            features.Animation = (imageFeatures & (1 << 1)) != 0;
+
+            // 3 reserved bytes should follow which are supposed to be zero.
+            stream.Read(buffer, 0, 3);
+            if (buffer[0] != 0 || buffer[1] != 0 || buffer[2] != 0)
+            {
+                WebpThrowHelper.ThrowImageFormatException("reserved bytes should be zero");
+            }
+
+            // 3 bytes for the width.
+            uint width = ReadUnsignedInt24Bit(stream, buffer) + 1;
+
+            // 3 bytes for the height.
+            uint height = ReadUnsignedInt24Bit(stream, buffer) + 1;
+
+            // Read all the chunks in the order they occur.
+            var info = new WebpImageInfo()
+            {
+                Width = width,
+                Height = height,
+                Features = features
+            };
+
+            return info;
+        }
+
+        /// <summary>
+        /// Reads a unsigned 24 bit integer.
+        /// </summary>
+        /// <param name="stream">The stream to read from.</param>
+        /// <param name="buffer">The buffer to store the read data into.</param>
+        /// <returns>A unsigned 24 bit integer.</returns>
+        public static uint ReadUnsignedInt24Bit(Stream stream, byte[] buffer)
+        {
+            stream.Read(buffer, 0, 3);
+            buffer[3] = 0;
+            return (uint)BinaryPrimitives.ReadInt32LittleEndian(buffer);
```

Comment:
Why do we cast here rather than using `ReadUInt32LittleEndian`?

Reply:
That was probably just a mistake: changed with 2a8418f9